website/blog/2023-11-26-Agent-AutoBuild/index.mdx

```diff
@@ -0,0 +1,152 @@
+```python
+builder.build(building_task, default_llm_config, coding=True)
+```
+If `coding` is not specified, AgentBuilder will determine on its own whether the user proxy should be added or not according to the task.
+
+### Step 5: execute the task
+Let agents generated in `build()` to complete the task collaboratively in a group chat.
+```python
+execution_task="Find a latest paper about gpt-4 on arxiv and find its potential applications in software."
+builder.start(task=execution_task)
+```
+
+### Step 6 (Optional): clear all agents and prepare for the next task
+You can clear all agents generated in this task by the following code if your task is completed or the next task is largely different from the current task.
+```python
+builder.clear_all_agents(recycle_endpoint=True)
+```
+If the agent's backbone is an open-source LLM, this process will also shutdown the endpoint server. More details in the next section.
+If necessary, you can use `recycle_endpoint=False` to retain the previous open-source LLM's endpoint server.
+
+## Save and Load
+You can save all necessary information of the built group chat agents by
+```python
+saved_path = builder.save()
+```
+Configs will be saved in the JSON format with following content:
+```json
+// FILENAME: save_config_TASK_MD5.json
+{
+    "building_task": "Find a paper on arxiv by programming, and analysis its application in some domain. For example, find a latest paper about gpt-4 on arxiv and find its potential applications in software.",
+    "agent_configs": [
+        {
+            "name": "...",
+            "model": "...",
+            "system_message": "..."
+        },
+        ...
+    ],
+    "manager_system_message": "...",
+    "coding": true,
+    "default_llm_config": {
+        "temperature": 0
+    }
+}
+```
+You can provide a specific filename, otherwise, AgentBuilder will save config to the current path with a generated filename `save_config_TASK_MD5.json`.
+
+You can load the saved config and skip the building process. AgentBuilder will create agents with those information without prompting the build manager.
+```python
+new_builder = AgentBuilder(config_path=config_path).load(saved_path)
+new_builder.start()
+```
+
+## Use Open-source LLM
+AutoBuild support open-source LLM by [vLLM](https://docs.vllm.ai/en/latest/index.html) and [FastChat](https://github.com/lm-sys/FastChat).
+Check the supported model list [here](https://docs.vllm.ai/en/latest/models/supported_models.html).
+After satisfying the requirements, you can add an open-source LLM's huggingface repository to the config file,
+```json,
+// Add the LLM's huggingface repo to your config file and use EMPTY as the api_key.
+[
+    ...
+    {
+        "model": "meta-llama/Llama-2-13b-chat-hf",
+        "api_key": "EMPTY"
+    }
+]
+```
+and specify it when initializing AgentBuilder.
+AgentBuilder will automatically set up an endpoint server for open-source LLM. Make sure you have sufficient GPUs resources.
```

Comment:
Does the AgentBuilder automatically choose which LLM to use or does user specify this?

Reply:
Currently, LLM is specified by user when creating an `AgentBuilder` instance. This may change in the future.